aggregator_api/src/routes.rs

```diff
@@ -93,9 +93,9 @@ pub(super) async fn post_task<C: Clock>(
     // TODO(#1524): clean this up with `aggregator_core::task::Task` changes
     // unwrap safety: this fake URL is valid
     let fake_aggregator_url = Url::parse("http://never-used.example.com").unwrap();
-    let aggregator_endpoints = match req.role {
-        Role::Leader => Vec::from([fake_aggregator_url, req.peer_aggregator_endpoint]),
-        Role::Helper => Vec::from([req.peer_aggregator_endpoint, fake_aggregator_url]),
+    let (leader_aggregator_endpoint, helper_aggregator_endpoint) = match req.role {
+        Role::Leader => (fake_aggregator_url, req.peer_aggregator_endpoint),
+        Role::Helper => (req.peer_aggregator_endpoint, fake_aggregator_url),
```

Comment:
While we're here, should we make both aggregator endpoints Option (or tagged enum, or even expand role into tagged enum `enum Role { Leader { endpoint: Url ...`) so that we don't have to carry around a fake aggregator endpoint? 

It does sound like a large yak-shave, so if it takes too much time or is too complex, we can leave it alone.

Reply:
I think the eventual cleanup is to make our view of tasks track what is needed by the particular role; specifically, a Leader aggregator needs to know only the Helper's aggregator endpoint, and a Helper doesn't need to know anyone's aggregator endpoint. This is tracked by #1524.

Reply:
Yes, #1524 will address this. That change will be pretty significant, so I don't want to do it here.